### ISSUE TITLE
Remove deprecated `mbstring.http_input` and `mbstring.http_output` from .htaccess

### ISSUE THREAD

**Issue by Egmund** [backdrop/backdrop-issues#6734]
## Description of the bug

I get there to lines in my error-log many times every day - seems every time someone accesses the site:
```
PHP Deprecated:  PHP Request Shutdown: Use of mbstring.http_output is deprecated in Unknown on line 0
PHP Deprecated:  PHP Request Shutdown: Use of mbstring.http_input is deprecated in Unknown on line 0
```
I have commented out (with #) the lines in .htaccess mentioning mbstring.http but the errorlog still get filled.

I cannot for sure say it is Backdrop core since it relates to "Unknown on line 0"

Backdrop CMS:                          1.29.1
Installation profile:                  standard
PHP version:                           8.2.24
Drupal 7 compatibility:                Tilsluttet
Database server:                       10.6.19-MariaDB-cll-lve
Web server:                            LiteSpeed
jQuery version:                        3.7.1
jQuery UI version:                     1.14.0
CKEditor 5 Version: 42.0.2             Used in formats: Basic

Temaer
======
Default theme:                         Basis (basis) 1.29.1
Admin theme:                           Seven* (seven) 1.29.1
                                       *used when editing or creating content

Enabled modules
===============

Kerne
-----
admin_bar                              1.29.1
block                                  1.29.1
ckeditor5                              1.29.1
color                                  1.29.1
comment                                1.29.1
config                                 1.29.1
contextual                             1.29.1
dashboard                              1.29.1
date                                   1.29.1
dblog                                  1.29.1
email                                  1.29.1
entity                                 1.29.1
field                                  1.29.1
field_sql_storage                      1.29.1
field_ui                               1.29.1
file                                   1.29.1
filter                                 1.29.1
image                                  1.29.1
installer                              1.29.1
language                               1.29.1
layout                                 1.29.1
link                                   1.29.1
list                                   1.29.1
locale                                 1.29.1
menu                                   1.29.1
node                                   1.29.1
number                                 1.29.1
options                                1.29.1
path                                   1.29.1
redirect                               1.29.1
search                                 1.29.1
system                                 1.29.1
taxonomy                               1.29.1
telemetry                              1.29.1
text                                   1.29.1
translation                            1.29.1
update                                 1.29.1
user                                   1.29.1
views                                  1.29.1
views_ui                               1.29.1

Contrib
-------
antibot                                1.x-1.2.1
antiscan                               1.x-1.0.12
contact_us_webform_recipe              1.x-1.0.0-beta05
digital_agency                         1.x-1.0.0-beta29
extlink                                1.x-1.18.2
faq_recipe                             1.x-1.0.8
faq_sample_content                     1.x-1.0.8
header_dropdown_menu                   1.x-1.0.0-alpha1
iivblock                               1.x-1.0.1
image_grid_block                       1.x-1.0.0-alpha02
ip_blocking                            1.x-1.0.18
mini_layouts                           1.x-1.0.3
nice_menus                             1.x-1.0.0
portfolio_projects_recipe              1.x-1.0.0-beta07
portfolio_projects_sample_content      1.x-1.0.0-beta07
services_recipe                        1.x-1.0.0-beta03
services_sample_content                1.x-1.0.0-beta03
testimonial_recipe                     1.x-1.0.8
testimonial_sample_content             1.x-1.0.8
views_bootstrap                        1.x-3.5.1
views_bootstrap_demo                   1.x-3.5.1
webform                                1.x-4.26.3
webform_validation                     1.x-1.16

**Comment by Egmund** [comment]
It might be in 'location' or 'language' modules, since I do not get these errors in an American English site where these modules are not activated.

**Comment by indigoxela** [comment]
@Egmund the PHP Deprecated messages you posted seem incomplete. I'd actually expect a hint on the calling code.

> It might be in 'locale' or 'language' modules

I doubt that. I'm successfully running sites with locale module turned on on PHP 8.3.
I suspect it's one of the contrib modules causing the nagging. But we'd need more info than "unknown on line 0". :wink: 

What is the error log (file) you're referring to? Is that an apache log? Or dblog? Something else?

Could you also do a quick check for "Logging and errors" settings on admin/config/development/logging?

And additionally your PHP settings re error logging (in your php.ini - location depends on your hosting/setup).

**Comment by Egmund** [comment]
This is from the error_log at root - usually public_html. Here is full content:
```
[18-Oct-2024 15:26:28 Europe/Copenhagen] PHP Deprecated:  PHP Request Shutdown: Use of mbstring.http_output is deprecated in Unknown on line 0
[18-Oct-2024 15:26:28 Europe/Copenhagen] PHP Deprecated:  PHP Request Shutdown: Use of mbstring.http_input is deprecated in Unknown on line 0
[18-Oct-2024 17:15:48 Europe/Copenhagen] PHP Deprecated:  PHP Request Shutdown: Use of mbstring.http_output is deprecated in Unknown on line 0
[18-Oct-2024 17:15:48 Europe/Copenhagen] PHP Deprecated:  PHP Request Shutdown: Use of mbstring.http_input is deprecated in Unknown on line 0
[19-Oct-2024 05:53:53 Europe/Copenhagen] PHP Deprecated:  PHP Request Shutdown: Use of mbstring.http_output is deprecated in Unknown on line 0
[19-Oct-2024 05:53:53 Europe/Copenhagen] PHP Deprecated:  PHP Request Shutdown: Use of mbstring.http_input is deprecated in Unknown on line 0
```
The server log have nothing regarding this.
'Logging and errors' is set as standard - I have not changed those.

**Comment by indigoxela** [comment]
@Egmund then let's try something else.

If I say "grep for mbstring" - does this sound familiar to you or am I talking a foreign language? :wink: 

My suggestion is something like: `grep -r mbstring modules/` - search for any of your contrib modules, that does something with "mbstring".
Otherwise I wouldn't know what to suggest with the limited info I have.

What sort of hosting is this? It might eventually also be that your php.ini (or other PHP setting) causes trouble here.

**Comment by Egmund** [comment]
I wouldn't know how to 'grep' on the shared host. But I do have cPanel.
I ill send this 'problem' to the hosting corp as well.

**Comment by Egmund** [comment]
It could be some temporary 'glitch'. There has been no further of these errors since early this morning.

**Comment by laryn** [comment]
Maybe we need to remove that deprecated php value in htaccess for everything except PHP5?

**Comment by Egmund** [comment]
Yes indeed Laryn. Since I commented them out - hardly any 'errors'.

**Comment by indigoxela** [comment]
> Since I commented them out - hardly any 'errors'.

That's actually the opposite of what you initially reported.

And this settings only applies to setups with mod_php. Probably that's the reason why I never saw the problem - different SAPI.

@Egmund can you check your Server API, please? Go to the status page on admin/reports/status, then click on the link "PHP-Information". On that phpinfo page search for "Server API". It should be quite on top of that page.

Is it really mod_php? Because if not - this part of htaccess would be totally irrelevant.
That whole section in htaccess (regardless of php version) will be ignored with fcgi or FPM/FastCGI.

If it's not mod_php, your problem must have been caused by something different.

**Comment by Egmund** [comment]
Thank you for taking time with this.
Server API 	LiteSpeed V8.1

Anything else I should look for in that 'report'?

After updating PHP version from 8.2 to 8.3 I have seen no errors.
Also; it could maybe be caused by moving the site somehow.

**Comment by indigoxela** [comment]
> Server API LiteSpeed

Ah, then these htaccess settings should be irrelevant. I'm no expert, but Litespeed doesn't read that AFAIK. It has its own (different) php handler.

> After updating PHP version from 8.2 to 8.3 I have seen no errors.

I'd interpret that as an additional evidence that - whatever may have happened - was something completely outside Backdrop.

**Comment by Egmund** [comment]
For the 'fun' of it: I un-commented the six lines in .htaccess - changed the file back to what BackdropCMS suggest - and I get six errors in the error_log:
```
[20-Oct-2024 11:23:56 Europe/Copenhagen] PHP Deprecated:  PHP Request Shutdown: Use of mbstring.http_output is deprecated in Unknown on line 0
[20-Oct-2024 11:23:56 Europe/Copenhagen] PHP Deprecated:  PHP Request Shutdown: Use of mbstring.http_input is deprecated in Unknown on line 0
[20-Oct-2024 11:24:00 Europe/Copenhagen] PHP Deprecated:  PHP Request Shutdown: Use of mbstring.http_output is deprecated in Unknown on line 0
[20-Oct-2024 11:24:00 Europe/Copenhagen] PHP Deprecated:  PHP Request Shutdown: Use of mbstring.http_input is deprecated in Unknown on line 0
[20-Oct-2024 11:24:00 Europe/Copenhagen] PHP Deprecated:  PHP Request Shutdown: Use of mbstring.http_output is deprecated in Unknown on line 0
[20-Oct-2024 11:24:00 Europe/Copenhagen] PHP Deprecated:  PHP Request Shutdown: Use of mbstring.http_input is deprecated in Unknown on line 0
```
These lines should definitely not be there.

**Comment by avpaderno** [comment]
LiteSpeed supports .htaccess files. That is what [WpW: .htaccess and Rewrite Rules reports](https://blog.litespeedtech.com/2017/07/26/wpw-rewrite-rules-in-the-proper-order/).

> LiteSpeed Web Server supports the Apache .htaccess file. This configuration file is important because it allows us to enable or disable additional functionality, and to define URL redirect rules, among other things.

**Comment by indigoxela** [comment]
> LiteSpeed supports .htaccess files.

Yes, it does. But these items are wrapped inside `<IfModule mod_php.c>` and I really doubt, that Litespeed considers these lines.
We have other folks using Backdrop with Litespeed and they never reported any problem. That's why I'm skeptical here.

**Comment by Egmund** [comment]
I tried disabling error logging at one site to see whether the errors are generated by the site (Backdrop, modules or themes). No errors generated. So, they come from the above-mentioned.
I am also confused. My previous host was also litespeed, and there were no errors.
Anyway: For me this issue is 'resolved' by commenting out the six lines in htaccess.

**Comment by quicksketch** [comment]
For reference here's an active discussion about removing `mbstring.http_output` and `mbstring.http_input` from PHP entirely, but looks like it's not going to happen until PHP 9: https://github.com/php/php-src/pull/12445

But `mbstring.http_output` and `mbstring.http_input` and [have been deprecated since PHP 5.6](https://wiki.php.net/rfc/remove_deprecated_functionality_in_php7). Since our minimum version is PHP 5.6, I think we can remove these lines from every version of PHP:

```
  php_value mbstring.http_input             pass
  php_value mbstring.http_output            pass
```

There's a related Drupal issue over at https://www.drupal.org/project/drupal/issues/2332295 (this change was already applied to Backdrop, and then the whole thing was dropped when we started requiring PHP 5.6).

**Comment by quicksketch** [comment]
I filed a PR at https://github.com/backdrop/backdrop/pull/4907 that simply removes these lines.

These options are deprecated and apparently have no effect on any version of PHP >= 5.6.

**Comment by indigoxela** [comment]
Given, that this does cause harm on some LiteSpeed setups, it's about time to remove these lines. RTBC :+1:

**Comment by laryn** [comment]
Thanks @Egmund for filing this, and @quicksketch for the PR. And thanks for the feedback @indigoxela and @avpaderno. Merged into 1.28.x and 1.x.
